### PR TITLE
Refactor authentication router to handle different user types

### PR DESCRIPTION
**Pull request by LEVIII007**
Description
In router/auth.py, the current implementation does not handle invalid user types (neither "user" nor "organization"). This can lead to errors during authentication.

Suggested Improvements
Return Error for Invalid Types:

If the user type is not "user" or "organization," return an error message.

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -25,11 +25,60 @@
 import string
 import time
 from logger_config import setup_logger
+from slowapi import Limiter
```
Comment (suryabulusu) [inline app/main.py:28]: hey whats this for in this PR

File: app/router/auth.py [hunk 2]
```diff
@@ -34,9 +40,8 @@ def create_access_token(auth_user: AuthUser, Authorize: AuthJWT = Depends()):
 
     elif auth_user.type == "user":
         if "is_user_valid" not in auth_user.dict().keys():
-            return HTTPException(
-                status_code=400,
-                detail="Data Parameter {} is missing!".format("is_user_valid"),
+            raise HTTPException(
```
Comment (suryabulusu) [inline app/router/auth.py:43]: no.. the old code is better formatted

File: app/router/auth.py [hunk 3]
```diff
@@ -14,16 +14,22 @@ def index():
...
+        raise HTTPException(
+            status_code=400,
+            detail="Invalid user type! Must be either 'user' or 'organization'.",
+        )
 
     if auth_user.type == "organization":
         if not auth_user.name:
-            return HTTPException(
-                status_code=400, detail="Data Parameter {} is missing!".format("name")
+            raise HTTPException(
```
Comment (suryabulusu) [inline app/router/auth.py:31]: old code is better formatted

File: app/router/auth.py [hunk 4]
```diff
@@ -55,10 +60,7 @@ def refresh_token(Authorize: AuthJWT = Depends()):
     Authorize.jwt_refresh_token_required()
     current_user = Authorize.get_jwt_subject()
     old_data = Authorize.get_raw_jwt()
-    if "group" in old_data:
-        custom_claims = {"group": old_data["group"]}
-    else:
-        custom_claims = {}
+    custom_claims = {"group": old_data.get("group", {})}
```
Comment (suryabulusu) [inline app/router/auth.py:63]: previous code is better formatted in terms of readability

File: app/main.py [hunk 5]
```diff
@@ -78,8 +127,11 @@ def authjwt_exception_handler(request: Request, exc: AuthJWTException):
     return JSONResponse(status_code=exc.status_code, content={"detail": exc.message})
 
 
+app.include_router(auth.router, prefix="/auth", tags=["authentication"])
```
Comment (suryabulusu) [inline app/main.py:130]: why is this better as compared to previous code? because of the tag?

**Review by suryabulusu — COMMENTED** [review]
left comments

File: app/main.py (hunk 1 above)
Comment (LEVIII007) [reply app/main.py:28]: slowapi is for applying rate limiting

**Pull request closed without merging**